install_scripts/templates/kubernetes/migrate.sh

```diff
@@ -0,0 +1,390 @@
+    logSuccess "Native app state saved"
+}
+
+checkOutput() {
+    if [ "$?" -ne 0 ]; then
+        cat "$1"
+        rm "$1"
+        exit 1
+    fi
+}
+
+# if all state needed from the native app has been saved, set the HAS_NATIVE_STATE flag to skip
+# starting the native app. This could follow an interrupted migration.
+checkNativeState() {
+    if [ ! -f "${TMP_DIR}/native.env" ]; then
+        return
+    fi
+    if [ ! -f "${TMP_DIR}/app-config.json" ]; then
+        return
+    fi
+    if [ ! -f "${TMP_DIR}/migration.json" ]; then
+        return
+    fi
+    if [ ! -f "${TMP_DIR}/retraced.sql" ]; then
+        return
+    fi
+    if [ ! -f "${TMP_DIR}/secrets.tar" ]; then
+        return
+    fi
+
+    HAS_NATIVE_STATE=1
+}
+
+# if Kubernetes is installed, set the HAS_KUBERNETES flag to avoid reinstalling
+checkKubernetes() {
+    if ! grep -q "kubectl exec" "/usr/local/bin/replicatedctl" 2>/dev/null ; then
+        return
+    fi
+    if ! isReplicatedctlReady ; then
+        return
+    fi
+    HAS_KUBERNETES=1
+}
+
+checkApp() {
+    set +e
+    if replicatedctl app status &>/dev/null ; then
+        HAS_APP=1
+    fi
+    set -e
+}
+
+waitNativeRetracedPostgresReady() {
+    set +e
+    for i in {1..30}; do
+        if docker exec retraced-postgres pg_isready -q 2>/dev/null ; then
+            set -e
+            return
+        fi
+        sleep 2
+    done
+    echo "Timedout waiting for native Retraced Postgres to become ready"
+    exit 1
+}
+
+restoreRetraced() {
+    spinnerPodRunning "default" "retraced-postgres"
+    # the server can take a bit to be ready for connections after the pod is running
+    set +e
+    for i in {1..30}; do
+        cat "${TMP_DIR}/retraced.sql" | kubectl exec $(kubectl get pods | grep retraced-postgres | awk '{ print $1 }') -- \
+            /bin/bash -c 'PG_DATABASE=$POSTGRES_DATABASE PGHOST=$POSTGRES_HOST PGUSER=$POSTGRES_USER PGPASSWORD=$POSTGRES_PASSWORD psql' &>/dev/null
+        if [ "$?" -eq 0 ]; then
+            set -e
+            return
+        fi
+        sleep 2
+    done
+    echo "Failed to restore audit log"
+    exit 1
+}
+
+restoreSecrets() {
+    local replPod=$(kubectl get pods --selector='app=replicated,tier=master' | tail -1 | awk '{ print $1 }')
+    if [ -z "$replPod" ]; then
+        echo "Cannot restore secrets: Replicated pod not found"
+        return
+    fi
+    kubectl cp "${TMP_DIR}/secrets.tar" "$replPod":/tmp/secrets.tar -c replicated
+    kubectl exec "$replPod" -c replicated -- tar -xf /tmp/secrets.tar -C /var/lib/replicated 
+}
+
+startAppOnK8s() {
+    logStep "Restoring app state"
+
+    logSubstep "wait for daemon pod to report ready"
+    waitReplicatedctlReady
+
+    logSubstep "restore audit log"
+    restoreRetraced
+
+    logSubstep "restore certificates"
+    restoreSecrets
+
+    logSubstep "restore console settings"
+    set +e
+    local needsActivation=0
+    replicatedctl migration import < "${TMP_DIR}/migration.json" 2>"${TMP_DIR}/import.txt"
```

Comment:
I couldn't figure out how to use redirection and pipe to grep in same command. Kept getting EOF errors, which is why I ended up sending to a file and grepping in a separate step.